bindings/java/src/test/java/org/apache/opendal/test/OperatorInputStreamTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+import org.apache.opendal.BlockingOperator;
+import org.apache.opendal.OperatorInputStream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class OperatorInputStreamTest {
+    @TempDir
+    private static Path tempDir;
+
+    @Test
+    void testReadWithInputStream() throws Exception {
+        final Map<String, String> conf = new HashMap<>();
+        conf.put("root", tempDir.toString());
+
+        try (final BlockingOperator op = BlockingOperator.of("fs", conf)) {
+            final String path = "OperatorInputStreamTest.txt";
+            final StringBuilder content = new StringBuilder();
+
+            final long multi = 1024;
+            for (long i = 0; i < multi; i++) {
+                content.append("[content] OperatorInputStreamTest\n");
+            }
+            op.write(path, content.toString().getBytes());
+
+            try (final OperatorInputStream is = OperatorInputStream.open(op, path)) {
```

Comment:
How about using an API like `op.getInputStream(path, options)` or `op.createInputStream(path, options)`? Which style is more ideal for Java users?

Reply:
This is possible and almost the same from my experience. There should not be any blocker from doing so.

The support of `option` can be a follow-up that would involved with a few things to convert data class cross language boundary.

Reply:
Ok, we can add `op.getInputStream` and options support later. Last thing, how about use `new` / `create` instead of `open`? I want to avoid making users feel like it's a file and there is an `open` syscall.

Reply:
You're right. Then I'd prefer to keep only `getInputStream` public.

Reply:
Updated.